src/network.c

```diff
@@ -195,6 +199,8 @@ gboolean download_mem(GBytes **data, const gchar *url, gsize limit, GError **err
 	GError *ierror = NULL;
 	char *dl_data = NULL;
 
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```

Comment:
Input checks for data and url should be added, too.

Reply:
Done.

Reply:
The check for `data` should be extended to:
``` c
g_return_val_if_fail(data && *data == NULL, FALSE);
```

Reply:
Done - I left that out intentionally as there are no explicit guarantees defined anywhere with respect to `*data`.